crates/node/src/shell/governance.rs

```diff
@@ -461,18 +467,41 @@ fn execute_pgf_steward_proposal<S>(
 where
     S: StorageRead + StorageWrite,
 {
-    for action in stewards {
-        match action {
-            AddRemove::Add(address) => {
-                pgf_storage::stewards_handle().insert(
-                    storage,
-                    address.to_owned(),
-                    StewardDetail::base(address),
-                )?;
-            }
-            AddRemove::Remove(address) => {
-                pgf_storage::stewards_handle().remove(storage, &address)?;
-            }
+    let maximum_number_of_pgf_steward_key =
+        pgf_storage::get_maximum_number_of_pgf_steward_key();
+    let maximum_number_of_pgf_steward = storage
+        .read::<u64>(&maximum_number_of_pgf_steward_key)?
+        .expect(
+            "Pgf parameter maximum_number_of_pgf_steward must be in storage",
+        );
+
+    // First, remove the appropriate addresses
+    for address in stewards.iter().filter_map(|action| match action {
+        AddRemove::Add(_) => None,
+        AddRemove::Remove(address) => Some(address),
+    }) {
+        pgf_storage::stewards_handle().remove(storage, address)?;
+    }
+
+    // Then add new addresses
+    let mut steward_count = pgf_storage::stewards_handle().len(storage)?;
+    for address in stewards.iter().filter_map(|action| match action {
+        AddRemove::Add(address) => Some(address),
+        AddRemove::Remove(_) => None,
+    }) {
+        #[allow(clippy::arithmetic_side_effects)]
+        if steward_count + 1 > maximum_number_of_pgf_steward {
```

Comment:
Should this be a checked operation?

Reply:
the limit will be a number < 100 and always > 0 so making the operation checked here is too much I think